src/skeleton/read.rs

```diff
@@ -89,14 +96,62 @@ pub(super) fn manifests<P: AsRef<Path>>(
                 &absolute_path
             )
         })?;
+        let mut targets = package.map(|p| gather_targets(p)).unwrap_or_default();
+        targets.sort_by(|a, b| a.path.cmp(&b.path));
+
         manifests.push(ParsedManifest {
             relative_path,
             contents: intermediate,
+            targets,
         });
     }
 
     Ok(manifests)
 }
+// I started implementing resolving targets through `cargo metadata`, but I wonder if it makes sense. We currently recreate
+fn gather_targets(package: &Package) -> Vec<Target> {
+    let manifest = package.manifest_path.clone().into_std_path_buf();
+    let root_dir = manifest.parent().unwrap();
+    package
+        .targets
+        .iter()
+        .filter_map(|target| {
```

Comment:
Isn't this just a `map`?

Reply:
It is, good catch. It was returning `None` before, but I changed it.